experiments/calibration/shared/rand_locations.jl

```diff
@@ -0,0 +1,39 @@
+using NCDatasets
+import EnsembleKalmanProcesses as EKP
+using ClimaUtilities.ClimaArtifacts
+import ClimaLand
+using StaticArrays
+
+# Returns a 2 SVectors of (lat, lon) tuples for n random locations on the land
+# surface. The two sets of locations are designed to be used as a training and
+# validation set.
+function rand_locations(surface_space, regridder_type, n = 100)
+    # Load the landsea mask data
+    datapath = ClimaLand.Artifacts.topmodel_data_path()
+    landsea_mask = SpaceVaryingInput(
+        datapath,
+        "landsea_mask",
+        surface_space;
+        regridder_type,
+    )
+
+    # Get the corresponding latitude and longitude values
+    lat = ClimaCore.Fields.coordinate_field(surface_space).lat
+    lon = ClimaCore.Fields.coordinate_field(surface_space).long
```

Comment:
You don't need to use ClimaCore Fields for this: you can directly look at the NetCDF file, which is more straightforward

Reply:
I agree that using the netcdf file of diagnostic output + mask, both from netcdf, would be simpler, and map directly to output data. 

As is, I dont understand how indices taken from the parent array of the coordinate space will map onto the diagnostics matrix, since those are not at exactly the same resolution. I dont think we will be getting the same locations in both cases. And likewise for ERA5 data, that will also need other indices.